openedx/core/djangoapps/safe_sessions/middleware.py

```diff
@@ -542,19 +570,20 @@ def _delete_cookie(request, response):
         )
 
 
-def log_request_user_changes(request):
+def track_request_user_changes(request):
     """
-    Instrument the request object so that we log changes to the `user` attribute. This is done by
-    changing the `__class__` attribute of the request object to point to a new class we created
-    on the fly which is exactly the same as the underlying request class but with an override for
-    the `__setattr__` function to catch the attribute chages.
+    Instrument the request object so that we store changes to the `user` attribute for future logging
+    if needed for debugging user mismatches. This is done by changing the `__class__` attribute of the request
+    object to point to a new class we created on the fly which is exactly the same as the underlying request class but
+    with an override for the `__setattr__` function to catch the attribute chages.
     """
     original_user = getattr(request, 'user', None)
 
     class SafeSessionRequestWrapper(request.__class__):
         """
         A wrapper class for the request object.
         """
+        user_changes = []
```

Comment:
Would calling this `safe_session_user_changes` make it simpler to track down what it is for if someone notices it on the session and wonders? Or, `debug_user_changes`? Just a thought. Feel free to disagree. :)